experiments/dsl/src/main/scala/org/apache/gearpump/streaming/dsl/StreamApp.scala

```diff
@@ -100,15 +97,30 @@ class TimeReplayableProducer[T:ClassTag](timeReplayableSource: TimeReplayableSou
 }
 
 class KafkaProducer[T:ClassTag](kafkaConfig: KafkaConfig, converter: Message => T) extends Traverse[T] {
-  val batchSize = kafkaConfig.getConsumerEmitBatchSize
-  val msgDecoder: MessageDecoder = kafkaConfig.getMessageDecoder
-  lazy val source = Some(new KafkaSource(kafkaConfig.getClientId, TaskId(0, 0), 1, kafkaConfig, msgDecoder)).map(kafkaSource => {
-    kafkaSource.startFromBeginning
+  lazy val context = new TaskContext {
+      override def taskId: TaskId = TaskId(0, 0)
+      override def appName: String = "gearpump"
+      override def schedule(initialDelay: FiniteDuration, interval: FiniteDuration)(f: => Unit): Cancellable = null
+      override def appId: Int = 0
+      override def executorId: Int = 0
+      override def parallelism: Int = 1
+      override def scheduleOnce(initialDelay: FiniteDuration)(f: => Unit): Cancellable = null
+      override def self: ActorRef = null
+      override def output(msg: Message): Unit = {}
+      override def upstreamMinClock: TimeStamp = Message.noTimeStamp
+      override def actorOf(props: Props): ActorRef = null
+      override def actorOf(props: Props, name: String): ActorRef = null
+      override def sender: ActorRef = null
+      override def appMaster: ActorRef = null
+      override def system: ActorSystem = null
+    }
```

Comment:
as discussed yesterday, is it possible to modify ```Traverse``` interface to pass TaskContext inside? So that we don't need to create a fake TaskContext object.